### PR TITLE
Set up github actions

### PR DESCRIPTION
**Pull request by bsamuel-ui**
Addressing #468, this replaces CircleCI with Github Actions. There are most likely more elegant ways to restructure the tests, but I wanted to do as much of a straight port as possible. Definitely view the diff with ignore whitespace. 😬

Much thanks to @AndrewFarley for assistance with this.

I've got it working for 2.7 and 3.6, and I think this is a good point to try and merge it. 🤞

Outstanding issues:

_dockerizePip fails on Windows._ tryBindPath is looking for the alpine docker image to run its test, and it doesn't find that architecture. At least that message is visible in debug logs now.

_Many 2.7 and 3.6 specific tests._ I think tests should all be python agnostic as much as humanly possible. That mostly means dropping a bunch of tests and inferring the correct setting for `--runtime`

_Tests do not need to be editing files._ There's a teardown method that was checking out a test file, that's just a bad idea. Serverless has a powerful configuration system, we should look that value up through an option.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module.exports = {
```
[the remark this answers is not in the document]
Comment (bsamuel-ui) [reply .eslintrc.js]: - [ ] Agreed, didn't think to look in package for this.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+        python-version: [3.6, 2.7]  # 3.7
+        node-version: [12]  # 8, 10
```
[the remark this answers is not in the document]
Comment (bsamuel-ui) [reply .github/workflows/test.yml]: - [ ] Create an issue for this

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
+name: Test
+
+on: [push]
+  # pull_request:
+  #   branches:
+  #     - master
+  # push:
+  #   branches:
+  #     - master
```
[the remark this answers is not in the document]
Comment (bsamuel-ui) [reply .github/workflows/test.yml]: Yeah, need to figure out what we want that behavior to be.

File: README.md [hunk 4]
```diff
@@ -1,11 +1,11 @@
 # Serverless Python Requirements
 
 [![serverless](http://public.serverless.com/badges/v3.svg)](http://www.serverless.com)
-[![CircleCI](https://circleci.com/gh/UnitedIncome/serverless-python-requirements.svg?style=shield)](https://circleci.com/gh/UnitedIncome/serverless-python-requirements)
+![Github Actions](https://github.com/UnitedIncome/serverless-python-requirements/workflows/Test/badge.svg)
 [![npm](https://img.shields.io/npm/v/serverless-python-requirements.svg)](https://www.npmjs.com/package/serverless-python-requirements)
 [![code style: prettier](https://img.shields.io/badge/code_style-prettier-ff69b4.svg)](https://github.com/prettier/prettier)
 
-A Serverless v1.x plugin to automatically bundle dependencies from
+A Serverless v5.x plugin to automatically bundle dependencies from
```
[the remark this answers is not in the document]
Comment (bsamuel-ui) [reply README.md]: Oh yeah, oops

**Comment by miketheman** [comment]
@bsamuel-ui It's been a while since this was reviewed-  have you had any time to make progress on this front? It's okay if you haven't - let me know, and I'd be open to continue with this.

File: .eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  extends: 'eslint:recommended',
+  globals: {
+    Atomics: 'readonly',
+    SharedArrayBuffer: 'readonly'
+  },
+  parserOptions: {
+    ecmaVersion: 2018
+  },
+  rules: {
+    'no-console': 0
```
[the remark this answers is not in the document]
Comment (bsamuel-ui) [reply .eslintrc.js:16]: Done.

File: .github/workflows/test.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (bsamuel-ui) [reply .github/workflows/test.yml:9]: It's going away.

File: package.json [hunk 6]
```diff
@@ -47,8 +47,9 @@
     "eslint": "^6.8.0",
     "prettier": "^1",
     "cross-spawn": "*",
-    "deasync-promise": "*",
-    "tape": "*"
+    "tape": "*",
+    "tape-promise": "*",
+    "lodash": "^4.16.15"
```
[the remark this answers is not in the document]
Comment (bsamuel-ui) [reply package.json:52]: It's used elsewhere, but I can compare to Object.entries in that case.

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+      - name: Install serverless
+        run: npm install -g serverless
+
+      - name: Install deps
+        run: npm install
+
+      - name: Test
+        run: npm run test
+        env:
+          USE_PYTHON: python
```
[the remark this answers is not in the document]
Comment (bsamuel-ui) [reply .github/workflows/test.yml:52]: So... that's a bit complicated. I'd like to chat about that on slack.

File: .github/workflows/test.yml [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+        run: python -m pip install --force setuptools wheel
+
+      - name: Install pipenv / poetry
+        run: python -m pip install pipenv poetry
+
+      - name: Install serverless
+        run: npm install -g serverless
+
+      - name: Install deps
+        run: npm install
```
[the remark this answers is not in the document]
Comment (bsamuel-ui) [reply .github/workflows/test.yml:47]: Nope, we'd have to add that as a separate action. Let's do that on another PR.

**Pull request merged**